src/figuresClasses.ts

```diff
@@ -1,11 +1,91 @@
-export interface Figure {}
+export interface Figure {
+  shape: `triangle` | `circle` | `rectangle`;
+  color: `red` | `green` | `blue`;
+  getArea: () => number;
+}
+
+export class Triangle implements Figure {
+  shape: `triangle`;
+
+  constructor(
+    public color: `red` | `green` | `blue`,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
```

Comment:
[CODE KNOWLEDGE] You don't need to manually assign constructor parameters to class properties when using access modifiers like `public`. TypeScript does this automatically.